assets/js/util/tracking/createEnableTracking.js

```diff
@@ -16,17 +16,22 @@ import { SCRIPT_IDENTIFIER, DATA_LAYER } from './constants';
 export default function createEnableTracking( config, dataLayerTarget ) {
 	const dataLayerPush = createDataLayerPush( dataLayerTarget );
 
+	let hasInsertedTag;
+
 	/**
-	 * Enables tracking by injecting the necessary script tag if not present.
+	 * Injects the necessary script tag if not present.
 	 */
-	return function enableTracking() {
-		config.trackingEnabled = true;
-
+	const initializeSnippet = () => {
 		const { document } = global;
-
-		if ( document.querySelector( `script[${ SCRIPT_IDENTIFIER }]` ) ) {
+		if ( undefined === hasInsertedTag ) {
+			hasInsertedTag = document.querySelector(
+				`script[${ SCRIPT_IDENTIFIER }]`
+			);
+		}
+		if ( hasInsertedTag ) {
 			return;
 		}
+		config.trackingEnabled = true;
```

Comment:
This function is only concerned with injecting the snippet, it should not change the `trackingEnabled` state – that's what `enableTracking` is for 🙂

Reply:
https://github.com/google/site-kit-wp/pull/4142/commits/4faffa2ddd8f92d4cf9aeae34ec7c9d2c948ce3f